libbeat/common/docker/helpers.go

```diff
@@ -0,0 +1,59 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package docker
+
+import (
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/safemapstr"
+)
+
+// ExtractContainerName strips the `/` characters that frequently appear in container names
+func ExtractContainerName(names []string) string {
+	output := names[0]
+
+	if len(names) > 1 {
+		for _, name := range names {
+			if strings.Count(output, "/") > strings.Count(name, "/") {
+				output = name
+			}
+		}
+	}
+	return strings.Trim(output, "/")
+}
+
+// DeDotLabels returns a new common.MapStr containing a copy of the labels
+// where the dots have been converted into nested structure, avoiding
+// possible mapping errors
+func DeDotLabels(labels map[string]string, dedot bool) common.MapStr {
```

Comment:
I wonder if this could be also reused in the dedotting we do in Kubernetes. (no need to change this here)

Reply:
Where I think that we could reuse it and is more related is here: https://github.com/elastic/beats/blob/16a646da88d835a4940de0e9d53b6f19b1f3704c/libbeat/processors/add_docker_metadata/add_docker_metadata.go#L190-L198

Where the code is basically the same.